### PR TITLE
Add additional config param in SyslogOutput Plugin for mutual TLS

### PR DESCRIPTION
**Pull request by Hisagar**
Update config parameters in SyslogOutput 
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
Following config parameters added in SyslogOutput Configuration for mutual TLS authentication
 - client_cert_path
 - private_key_path
 - private_key_passphrase
 - enable_system_cert_store
 - allow_self_signed_cert
 - verify_fqdn
 - fqdn
 - version

### Why?
To add support for mutual TLS for syslog output using above mentioned config parameters

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

File: pkg/sdk/logging/model/output/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -1403,6 +1403,16 @@ func (in *SyslogOutputConfig) DeepCopyInto(out *SyslogOutputConfig) {
 		*out = new(secret.Secret)
 		(*in).DeepCopyInto(*out)
 	}
+	if in.PrivateKeyPath != nil {
```
Comment (alexxed) [inline pkg/sdk/logging/model/output/zz_generated.deepcopy.go:1406]: you added 5 new fields, but this file has only two, is it correct?

Comment (Hisagar) [reply pkg/sdk/logging/model/output/zz_generated.deepcopy.go:1406]: This file is autogenerated using `make generate`  command

File: docs/configuration/crds/extensions/v1alpha1/eventtailer_types.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
+---
+title: EventTailer
+weight: 200
+generated_file: true
+---
+
+## EventTailerSpec
```
Comment (alexxed) [inline docs/configuration/crds/extensions/v1alpha1/eventtailer_types.md:7]: this is unrelated to the pr, remove it

**Review by alexxed — COMMENTED** [review]
Please remove all the extra new docs you added. Keep in the PR only what's related to the PR. There is way too much noise, your PR has 9 thousand lines.

File: fluentd-image/v1.14/Dockerfile [hunk 3]
```diff
@@ -73,7 +73,7 @@ RUN apk update \
          fluent-plugin-sqs \
          fluent-plugin-kube-events-timestamp \
  && gem specific_install -l https://github.com/tarokkk/fluent-plugin-logzio.git \
- && gem specific_install -l https://github.com/cloudfoundry/fluent-plugin-syslog_rfc5424.git \
+ && gem specific_install -l https://github.com/Hisagar/fluent-plugin-syslog_rfc5424.git \
```
[the remark this answers is not in the document]
Comment (Hisagar) [reply fluentd-image/v1.14/Dockerfile]: Updated 👍

File: fluentd-image/v1.14/Dockerfile [hunk 4]
```diff
@@ -1,5 +1,5 @@
 FROM alpine:3.14
-LABEL Description="Fluentd docker image" Vendor="Banzai Cloud" Version="1.14.4"
+LABEL Description="Fluentd docker image" Vendor="Banzai Cloud" Version="1.14.5"
```
Comment (alexxed) [inline fluentd-image/v1.14/Dockerfile:2]: why did you change this? in logging_types below it's still 1.14.4

File: pkg/sdk/logging/api/v1beta1/logging_types.go [hunk 5]
```diff
@@ -115,8 +115,8 @@ type DefaultFlowSpec struct {
 const (
 	DefaultFluentbitImageRepository         = "fluent/fluent-bit"
 	DefaultFluentbitImageTag                = "1.8.9"
-	DefaultFluentdImageRepository           = "ghcr.io/banzaicloud/fluentd"
-	DefaultFluentdImageTag                  = "v1.14.4-alpine-1"
+	DefaultFluentdImageRepository           = "672327909798.dkr.ecr.us-east-1.amazonaws.com/sagar-fluentd"
```
Comment (alexxed) [inline pkg/sdk/logging/api/v1beta1/logging_types.go:118]: we need to check with banzai, but I see this image is built only when tagged https://github.com/banzaicloud/logging-operator/blob/b2387f8d9445c01c0c23c119785949e8c38208f7/.github/workflows/docker_fluentd.yml, so we might need 2 prs, one for fluentd changes, and another one that uses the fluentd changes

File: docs/configuration/crds/extensions/v1alpha1/eventtailer_types.md (hunk 2 above)
Comment (Hisagar) [reply docs/configuration/crds/extensions/v1alpha1/eventtailer_types.md]: Acknowledged

File: pkg/sdk/logging/api/v1beta1/logging_types.go (hunk 5 above)
Comment (Hisagar) [reply pkg/sdk/logging/api/v1beta1/logging_types.go:118]: But to push changes in ghcr.io/banzaicloud/fluentd we don’t have access right? how we are going to push this changes?

File: fluentd-image/v1.14/Dockerfile (hunk 4 above)
Comment (Hisagar) [reply fluentd-image/v1.14/Dockerfile:2]: yes, this need to revert 👍

File: pkg/sdk/logging/api/v1beta1/logging_types.go (hunk 5 above)
Comment (ahma) [reply pkg/sdk/logging/api/v1beta1/logging_types.go]: @Hisagar please update the pr, and set the DefaultFluentdImageTag -> v1.14.4-alpine-2. If everything looks good I'll tag the fluentd on this branch.

**Review by razee-khan-aq — APPROVED** [review]
Looks good to me.

**Comment by Hisagar** [comment]
> @Hisagar please resolve the conflict as well. Thanks

@ahma conflicts resolved, Thanks :)

**Review by ahma — APPROVED** [review]
Thanks @Hisagar, We'll fix the rest with the next pr
